Refactor Beads tab

priyashuu opened the pull request:
**PR Description**

This PR is opened to make the **Beads** tab fully responsive across devices, ensuring a smooth and consistent experience for desktop, tablet, and mobile users.

### **Enhancements & New Features**

1. **Pool Ranking (1 Week)**

   * Added a **weekly pool ranking table** with detailed metrics:

     * Recent Block ,Hashrate .Blocks mined ,Average Health ,Average Block Fee ,Empty Blocks count

2. **Reward Tab Improvements**

   * Expanded reward analytics   (with BTC and USD values) to display data for:

     * **Last Hour**
     * **Last Week**
     * **Last Month**
   * Enhanced **Average Per Block** statistics
   * Added a **BTC vs USD rewards comparison graph** for better visual insights

3. **Difficulty Graph**

   * Introduced a **Network Difficulty** chart with real-time tracking
   * Displays current difficulty in **Tera** units with timestamped tooltips

<img width="1805" height="807" alt="image" src="https://github.com/user-attachments/assets/2f057cb5-16f6-4a91-b1e7-6be17b46ef67" />
<img width="797" height="643" alt="image" src="https://github.com/user-attachments/assets/a865dbeb-a658-429a-ae4e-6f8b1da311e4" />
<img width="1759" height="810" alt="image" src="https://github.com/user-attachments/assets/f4ac2618-fd95-424b-9984-844e9b4fc7dd" />
<img width="1821" height="814" alt="image" src="https://github.com/user-attachments/assets/4ad34402-98c0-42da-b9ef-ed1ed7cfc45c" />
<img width="769" height="702" alt="image" src="https://github.com/user-attachments/assets/f8fee9bb-41fe-4724-8ce2-8067cfd4f862" />

mcelrath committed:
doc: add contribution guidelines (#156) (#157)

Co-authored-by: Mohd Zaid <72139621+zaidmstrr@users.noreply.github.com>

u32luke committed:
Update general_considerations.md (#173)

Typo fix

priyashuu committed:
Merge remote-tracking branch 'origin/dev' into refactor

priyashuu committed:
Refactor: Remove generateRewardHistory function and its tests

- Deleted the generateRewardHistory function from generateRewardHistory.tsx.
- Removed associated tests from generateRewardHistory.test.tsx.
- Updated RewardTab to reflect the removal of the reward history generation logic.

UI: Update styles in HashrateTab, LatencyTab, and TransactionsTab

- Adjusted spacing and background color in HashrateTab, LatencyTab, and TransactionsTab components for improved UI consistency.

Feature: Add Pool Dominance component with WebSocket integration

- Implemented PoolDominance component to visualize pool dominance data using a pie chart.
- Integrated WebSocket for real-time updates on pool data.
- Added utility functions for formatting hashrate values.

Feature: Introduce Difficulty component for network difficulty visualization

- Created Difficulty component to display network difficulty over time.
- Integrated WebSocket for real-time difficulty updates.
- Utilized AdvancedChart for rendering difficulty data.

Refactor: Clean up Dashboard component and remove Installation instructions

- Removed InstallationInstructions component and its associated tests.
- Updated Dashboard component to eliminate references to the removed InstallationInstructions.
- Cleaned up unused imports and code related to the installation process.

Enhancement: Update constants and types for better clarity

- Modified Constants.ts to include new tabs and color constants.
- Updated Types.ts to refine reward and pool data structures.

priyashuu committed:
Refactor: Update environment variables, clean up BeadRow and PoolDominance components, and enhance RewardsDashboard WebSocket handling

priyashuu committed:
Refactor: Enhance fetchPoolInfo logging, clean up PoolDominance WebSocket handling, and implement StatCard component for RewardsDashboard

priyashuu committed:
Refactor: Enhance fetchPoolInfo to include latest block height and pool link, update PoolDominance component to display new data, and adjust TransactionList layout for better responsiveness

priyashuu committed:
Merge remote-tracking branch 'origin/dev' into refactor

priyashuu committed:
Refactor: Clean up code formatting and remove unused constants in various components

priyashuu committed:
Refactor: Improve layout responsiveness in BeadRow and MinedSharesExplorer components, and clean up tsconfig.json library settings

priyashuu committed:
Refactor: Update PoolDominance component layout and improve responsiveness, enhance test cases for AnimatedStatCard and BeadRow, and remove unused constants from Constants.ts

priyashuu committed:
prettier formatting

priyashuu committed:
Test: Add unit tests for fetchReward function to validate block history and error handling

priyashuu committed:
Refactor: Update grid layouts and improve responsiveness in BeadRow, MinedSharesExplorer, and TransactionList components; enhance styling in RewardsSection and Difficulty components

priyashuu committed:
Refactor: Update Jest configuration and enhance tests for TransactionList and BeadRow components; improve transaction rendering and interaction handling

priyashuu committed:
Refactor: Rewrite tests for TransactionsTab component; enhance websocket handling, loading states, and error management

its-me-abhishek commented on dashboard/api/utils/__tests__/fetchBlockDetails.test.js:
```diff
@@ -17,6 +17,7 @@ describe('fetchBlockDetails', () => {
 
     mockClient = {
       readyState: WebSocket.OPEN,
+      OPEN: WebSocket.OPEN,
```
should've used lowercase (edit: if possible)

priyashuu replied:
I’ll keep this as is for now, since changing OPEN here would break functionality in other files that depend on it.

its-me-abhishek commented on dashboard/api/utils/__tests__/fetchBlockDetails.test.js:
```diff
@@ -69,7 +85,7 @@ describe('fetchBlockDetails', () => {
   });
 
-  it('should not fetch again if block hash hasn’t changed', async () => {
+  it('should not fetch again if block hash hasnt changed', async () => {
```
small typo, missed an apostrophe

its-me-abhishek commented on dashboard/api/utils/fetchBlockDetails.js:
```diff
@@ -107,7 +111,7 @@ export async function fetchBlockDetails(wss) {
         txCount: blockData.tx.length,
         nonCoinbaseTxCount: blockData.tx.length - 1,
-        reward: rewardBTC,
+        reward,
```
please use destructuring here as well

its-me-abhishek commented on dashboard/tsconfig.json:
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "lib": ["dom", "dom.iterable", "esnext", "ES2020", "DOM", "DOM.Iterable"],
```
why were these removed

priyashuu replied:
I removed the extra lib entries because they were duplicates or already included by default with our current target. The DOM types are still included, so we haven’t lost any browser API support. This just makes the config cleaner and avoids potential type conflicts.

its-me-abhishek reviewed (changes requested):
Please fix

priyashuu committed:
Refactor: Improve block data handling in fetchBlockDetails function; enhance variable destructuring for clarity

priyashuu committed:
Refactor: Update environment variable naming for clarity; enhance WebSocket cleanup in Difficulty component and improve conditional rendering in TrendsTab and MinedSharesExplorer components

priyashuu committed:
used formatFeepercentage function to avoid confusion ,

priyashuu committed:
updated readme file and env example

mcelrath committed:
Merge branch 'dev' into refactor

priyashuu committed:
Merge remote-tracking branch 'origin/dev' into refactor

priyashuu committed:
fixed the imports of the inventory tab in the dashboard file

priyashuu committed:
removed the `node-fetch` dependency s we no longer neede

priyashuu committed:
fix: replace fragile timestamp parsing with ISO-based parsing

- Updated frontend `parseTimestamp` to use ISO strings returned from backend
- Removed manual string splitting logic (DD/MM/YYYY assumptions) , used formatfeePercentage in pool dominance for cal avg block fees to remove the dupilcacy ,added the timestamp also to the tooltip in the rewards graph

priyashuu committed:
now tooltip show only 2 decimal values by using toFixed(2)

priyashuu committed:
added tests for the reward section and poolinfo

priyashuu committed:
fixes the failed tests

priyashuu committed:
Merge remote-tracking branch 'origin/dev' into refactor

priyashuu committed:
resolved conflicts and prettier formating

priyashuu committed:
removed the duplicate moduleNameMapper from the jest conf

priyashuu committed:
prettier formatting

priyashuu commented on dashboard/src/components/BeadsTab/PoolDominance/PoolDominance.tsx:
```diff
@@ -0,0 +1,210 @@
+
+      <div className="mt-6">
+        <div className="w-full">
+          <div className="sm:hidden max-md:hidden  lg:block ">
```
I tried that, but it didn’t work though

priyashuu commented on dashboard/src/components/BeadsTab/BeadRow.tsx:
```diff
@@ -23,24 +21,24 @@ export default function BeadRow({
+        font-medium font-mono`}
           >
-            {bead.name}
+            {bead.name.replace(/^#/, '')}
```
The regex replacement bead.name.replace(/^#/, '') is only used in this file, so extracting it into a separate utility function may be unnecessary. Keeping it inline here makes the code simpler and avoids extra indirection.

mcelrath reviewed (approved):
Looks good!
